core/frontend/src/store/beacon.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  Action,
+  getModule, Module, Mutation, VuexModule,
+} from 'vuex-module-decorators'
+
+import Notifier from '@/libs/notifier'
+import store from '@/store'
+import { Domain } from '@/types/beacon'
+import { beacon_service } from '@/types/frontend_services'
+import back_axios, { backend_offline_error } from '@/utils/api'
+import { callPeriodically } from '@/utils/helper_functions'
+
+const notifier = new Notifier(beacon_service)
+
+let prefetched_domains = false
+let prefetched_ips = false
+
+@Module({
+  dynamic: true,
+  store,
+  name: 'beacon',
+})
+
+class BeaconStore extends VuexModule {
+  API_URL = '/beacon/v1.0'
+
+  available_domains: Domain[] = []
+
+  listeners_count = 0
+
+  client_ip_address = ''
+
+  nginx_ip_address = ''
+
+  @Mutation
+  setAvailableDomains(domains: Domain[]): void {
+    this.available_domains = domains
+  }
+
+  @Mutation
+  setClientIpAddress(ip: string): void {
+    this.client_ip_address = ip
+  }
+
+  @Mutation
+  setNginxIpAddress(ip: string): void {
+    this.nginx_ip_address = ip
+  }
+
+  @Mutation
+  addListener(): void {
+    this.listeners_count += 1
+  }
+
+  @Mutation
+  removeListener(): void {
+    if (this.listeners_count <= 0) {
+      console.error('Deregister called with zero listeners.')
+      return
+    }
+    this.listeners_count -= 1
+  }
+
+  @Action
+  async fetchAvailableDomains(): Promise<void> {
+    if (prefetched_domains && !this.listeners_count) {
+      return
+    }
+    back_axios({
+      method: 'get',
+      url: `${this.API_URL}/services`,
+      timeout: 10000,
+    })
+      .then((response) => {
+        const available_domains = response.data
+        this.setAvailableDomains(available_domains)
+        prefetched_domains = true
+      })
+      .catch((error) => {
+        this.setAvailableDomains([])
+        if (error === backend_offline_error) { return }
+        const message = error.response.data.detail ?? error.message
+        notifier.pushBackError('BEACON_DOMAINS_FETCH_FAIL', message)
+      })
```

Comment:
No need for line 81 and 82. They are what differs `pushError` from `pushBackError`, so you only need to do `pushBackError('BEACON...', error)`.